sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -268,30 +268,32 @@ case class AlterViewAsCommand(
     name: TableIdentifier,
     originalText: String,
     query: LogicalPlan) extends RunnableCommand {
+  require(query.resolved)
 
   import ViewHelper._
 
   override def innerChildren: Seq[QueryPlan[_]] = Seq(query)
 
   override def run(session: SparkSession): Seq[Row] = {
-    // If the plan cannot be analyzed, throw an exception and don't proceed.
-    val qe = session.sessionState.executePlan(query)
-    qe.assertAnalyzed()
-    val analyzedPlan = qe.analyzed
```

Comment:
`query` will be an analyzed plan since `AlterViewAs` is converted to `AlterViewAsCommand` in Analyzer phase (`ResolveSessionCatalog`).

Reply:
shall we remove the similar code in `CreateViewCommand`?

Reply:
We cannot yet because of the following: https://github.com/apache/spark/blob/f340857757034ac955862b34e60322ca5ee81758/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/CacheTableExec.scala#L96-L105

Here, `query` is not resolved yet. I will get to this though.